### ISSUE TITLE
What is the meaning of `readOnly` in a definition?

### ISSUE THREAD

**Issue by itowlson** [deislabs/cnab-spec#247]
The typical usage of `readOnly` in JSON Schema, as far as I can tell, is for REST-type scenarios in which a particular attribute can only be _read_, not _written_.  It allows a schema to be written for the resource rather than separate schemas for reading and writing.  An example is a Person resource where a `GET /person/itowlson` might include a `lastLoginTime`, but `PUT /person/itowlson` should NOT include `lastLoginTime`.

It is not clear how this maps to CNAB parameters or outputs.

* Parameters must surely be settable.  Is `readOnly` on a parameter saying that it always has its default value and cannot be set by tooling (e.g. cannot be set using `duffle --set`)?  Is it saying that the parameter may be set on initial install but not _subsequently_ modified?  The latter is implied by #229 but is significantly at odds with typical JSON Schema usage and might be better handled by `"applyTo": ["install"]`...
* Outputs seem inherently `readOnly` - is it a no-op to apply it to an output?

**Comment by youreddy** [comment]
It's important to note that `readOnly` and `writeOnly` are annotations in JSON schema and that a runtime shouldn't throw a _schema_ validation error if the annotation is violated. Which is unfortunate because it means the user who configured a `readOnly` param won't get useful feedback on why their config was not applied.  

Which is why I agree that behavior should be further clarified in the spec. I would be comfortable saying that setting `readOnly` parameter can be set on initial install but not subsequently modified.  This [excerpt](https://json-schema.org/latest/json-schema-validation.html#rfc.section.10.3) from the JSON schema makes it feel like it's okay to do that: 

```
If "readOnly" has a value of boolean true, it indicates that the value of the instance is managed exclusively by the owning authority, and attempts by an application to modify the value of this property are expected to be ignored or rejected by that owning authority.
```

I'm not in favor of having `readOnly` mean that the parameter has a default value that cannot be changed just because it defeats the purpose of defining the parameter in the first place.

We should also consider having a `SHOULD` statement in the spec that encourages runtime authors to throw an error or warning if a `readOnly` param is set.

**Comment by jeremyrickard** [comment]
Closed by #265
